rest/admin_api.go

```diff
@@ -574,8 +574,8 @@ func (h *handler) handleSGCollect() error {
 		return base.HTTPErrorf(http.StatusBadRequest, "Unable to parse request body: %v", err)
 	}
 
-	if errs := params.Validate(); len(errs) > 0 {
-		return base.HTTPErrorf(http.StatusBadRequest, "Invalid options used for sgcollect_info: %#v", errs)
+	if errs := params.Validate(); errs.ErrorOrNil() != nil {
```

Comment:
Wouldn't it be preferable for params.Validate() to return ErrorOrNil(), instead of relying on the caller to invoke it?  That seems like the recommended usage based on the MultiError docs.

Reply:
Yeah, that is a fair point. In the old code, a multierror.Error was returning from params.Validate(). But this has been changed to return an error from params.Validate() instead. And we can directly check error is not nil.